packages/cli/src/lifecycles/graph.js

```diff
@@ -227,66 +206,100 @@ const generateGraph = async (compilation) => {
+               * title: used to customize the <title></title> tag of the page, inferred from the filename
+               * route: URL for accessing the page from the browser
+               * layout: the custom layout of the page
+               * data: custom page frontmatter
+               * imports: per page JS or CSS file imports specified from frontmatter
+               * resources: all script, style, etc resources for the entire page as URLs
+               * outputHref: href to the file in the output folder
+               * pageHref: href to the page's filesystem file
+               * isSSR: if this is a server side route
+               * prerender: if this page should be statically exported
+               * isolation: if this page should be run in isolated mode
+               * hydration: if this page needs hydration support
+               * servePage: signal that this is a custom page file type (static | dynamic)
+               */
+              const page = {
+                id: getIdFromRelativePathPath(relativePagePath, extension),
+                label,
+                title,
+                route: `${basePath}${route}`,
+                layout,
                 data: customData || {},
-                filename,
-                id,
-                relativeWorkspacePagePath: relativePagePath,
-                label: id.split('-')
-                  .map((idPart) => {
-                    return `${idPart.charAt(0).toUpperCase()}${idPart.substring(1)}`;
-                  }).join(' '),
                 imports,
                 resources: [],
-                outputPath: route === '/404/'
-                  ? '/404.html'
-                  : `${route}index.html`,
-                path: filePath,
-                route: `${basePath}${route}`,
-                layout,
-                title,
+                pageHref: new URL(relativePagePath, pagesDir).href,
+                outputHref: route === '/404/'
+                  ? new URL('./404.html', outputDir).href
+                  : new URL(`.${route}index.html`, outputDir).href,
+                // outputPath: route === '/404/'
+                //   ? '/404.html'
+                //   : `${route}index.html`,
                 isSSR: !isStatic,
                 prerender,
                 isolation,
                 hydration,
                 servePage: isCustom
-              });
+              };
+
+              pages.push(page);
+
+              // handle collections
+              const pageCollection = customData.collection;
+
+              if (pageCollection) {
+                if (!collections[pageCollection]) {
+                  collections[pageCollection] = [];
+                }
+
+                collections[pageCollection].push(page);
+              }
+
+              compilation.collections = collections;
             } else {
               console.debug(`Unhandled extension (${extension}) for route => ${route}`);
```

Comment:
[Noticed in this PR](https://github.com/ProjectEvergreen/greenwood/pull/1285/commits/a5ac177ca9edec948c289cec8ee19305db452cc8), we has a couple issues here
1. `route` is undefined
    ```js
    console.debug(`Unhandled extension (${extension}) for page => ${filename}`);
    ```
1. we should probably ignore hidden files
    ```sh
    building from local sources...
    { extension: '.DS_Store' }
    file:///Users/owenbuckley/Workspace/project-evergreen/greenwood/packages/cli/src/lifecycles/graph.js:292
                  console.debug(`Unhandled extension (${extension}) for route => ${route}`);
    ```